examples/custom_404.rs

```diff
@@ -0,0 +1,38 @@
+extern crate iron;
+extern crate router;
+
+// To build, $ cargo test
+// To use, go to http://127.0.0.1:3000/test
+
+use iron::{Iron, Request, Response, IronResult, AfterMiddleware, Chain};
+use iron::error::{IronError};
+use iron::status;
+use router::{Router};
+
+struct Custom404;
+
+impl AfterMiddleware for Custom404 {
+
+    fn catch(&self, _: &mut Request, err: IronError) -> IronResult<Response> {
+        println!("Hitting custom 404 middleware");
+
+        match err.response.status {
```

Comment:
When I try `use router::{Router, NoRoute}` so I have NoRoute in scope to use this technique, the compiler tells me "unresolved import `router::NoRoute`. There is no `NoRoute` in `router`". I think hitting this error is why I tried to match on err.response.status to begin with.